distributed/sizeof.py

```diff
@@ -11,6 +16,11 @@ def safe_sizeof(obj, default_size=1e6):
     This returns a default size of 1e6 if the sizeof function fails
     """
     try:
+        # Handle zero-strided NumPy arrays
+        if np is not None and isinstance(obj, np.ndarray) and 0 in obj.strides:
+            from .protocol.numpy import _zero_strided_slice
```

Comment:
It seems off to include this special case in this function.  Is this a case that we can fix upstream?

Reply:
I agree this seems a little off. My main motivation for including this special case here instead of upstream in `dask.sizeof` was because this is tied specifically to how `distributed` handles zero-strided arrays. In theory, `dask.sizeof` could be used independently of `distributed`. Although I'm not sure how common that is (I suspect probably not very common).

We could move this upstream to `dask`. Does that seem more reasonable than what I have here?

Reply:
To be honest I'm not sure what you mean by zero-strided numpy array.  This is an array that is stored as, say a 2d array, but uses stride tricks to appear to be 3d or 4d or somethiing?

In that case does `dask.sizeof` report the correct size of the object in memory?  I guess I'm not sure how what distributed does here affects things.  If we have one of these objects in memory, how much RAM does it take up?  I would think that this would be independent of which library is asking for the size.